compiler/rustc_target/src/target_features.rs

```diff
@@ -328,6 +355,7 @@ const X86_ALLOWED_FEATURES: &[(&str, Stability, ImpliedFeatures)] = &[
     ("sha512", Unstable(sym::sha512_sm_x86), &["avx2"]),
     ("sm3", Unstable(sym::sha512_sm_x86), &["avx"]),
     ("sm4", Unstable(sym::sha512_sm_x86), &["avx2"]),
+    ("soft-float", Forbidden, &[]), // changes float ABI
```

Comment:
We also have a stable `-Csoft-float` argument that is documented as using the soft float ABI for arm hard-float targets.

Reply:
Uh... okay that's an odd choice. That's, like, very unsound, right? If you use the shipped standard library and that flag, you're going to get some fun explosions. How did people imagine that would work? The docs don't even say anything like "THIS IS VERY UNSOUND MAKE SURE YOU KNOW WHAT YOU ARE DOING".

Everywhere else we use separate softfloat/hardfloat targets to ensure ABI correctness.

Reply:
Yeah, I agree. That flag shouldn't have existed.

Reply:
Luckily it seems like there are only a couple of uses of that flag on github.

Reply:
Let's make that a new issue, it's related to this PR but clearly needs a separate discussion and fix: https://github.com/rust-lang/rust/issues/129893.